### PR TITLE
Improve security group validation, add --no-validate option

### PR DESCRIPTION
**Pull request by bburtin**
When the subnet isn't specified, make sure that the specified security
groups are in the default VPC.  Add a get_default_vpc() method to
AWSService.

Replace the --validate-ami and --no-validate-ami options with a new
--no-validate option that controls validation of AMIs, subnets, and
security groups.  Hide the old options for the short term so that we
don't break backward compatibility.

Add support for a default VPC to the test framework.  Create test
security groups in that VPC when the VPC id is not specified.

### CODE REVIEW

File: brkt_cli/aws_service.py [hunk 1]
```diff
@@ -529,6 +533,12 @@ def detach_volume(self, vol_id, instance_id=None, force=True):
     def attach_volume(self, vol_id, instance_id, device):
         return self.conn.attach_volume(vol_id, instance_id, device)
 
+    def get_default_vpc(self):
+        vpcs = self.conn.get_all_vpcs(filters={'is-default': 'true'})
```
Comment (rohang06) [inline brkt_cli/aws_service.py]: is it possible to return more than one vpc here? i.e. a user misconfiguration

File: brkt_cli/__init__.py [hunk 2]
```diff
@@ -59,7 +59,27 @@ def _validate_subnet_and_security_groups(aws_svc,
...
 
-    return len(vpc_ids) <= 1
+    if len(vpc_ids) > 1:
+        raise ValidationError(
+            'Subnet and security groups must be in the same VPC.')
+
+    if not subnet_id and vpc_ids:
+        (vpc_id, ) = vpc_ids
+        default_vpc = aws_svc.get_default_vpc()
+        log.debug(
```
Comment (rohang06) [inline brkt_cli/__init__.py]: check for default_vpc being None? And are we concerned about the scenario where default_vpc and vpc_id are both None?

File: brkt_cli/aws_service.py (hunk 1 above)
Comment (bburtin) [inline brkt_cli/aws_service.py]: Yeah, but the code will still work.  We'll return the first element in the list.

File: brkt_cli/__init__.py (hunk 2 above)
Comment (bburtin) [inline brkt_cli/__init__.py]: I check for it on the next line.

**Comment by rohang06** [comment]
Cool, lgtm

**Comment by bburtin** [comment]
I'm going to add some comments to address Rohan's questions and will open a new pull request.

**Pull request closed without merging**